lib/ssl/src/tls_gen_connection.erl

```diff
@@ -325,6 +325,38 @@ handle_info({CloseTag, Socket}, StateName,
             %% is called after all data has been deliver.
             {next_state, StateName, State#state{protocol_specific = PS#{active_n_toggle => true}}, []}
     end;
+handle_info(ktls_next_record, StateName,
+            #state{protocol_buffers = #protocol_buffers{tls_cipher_texts = []},
+                   protocol_specific = #{ktls_inactive := true, size := Size} = ProtocolSpecific,
+                   static_env = #static_env{socket = Socket,
+                                            data_tag = Protocol,
+                                            close_tag = CloseTag,
+                                            transport_cb = Transport}} = State) ->
+    % Read next ssl record in socket
+    case Size of
+        -1 ->
+            case Transport:recv(Socket, 5, 1000) of
```

Comment:
I think the inet driver actually has a TLS packet mode and using that with active once would be a nicer approach I think!